src/librustc_trans/debuginfo/metadata.rs

```diff
@@ -657,6 +657,15 @@ pub fn type_metadata<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
     metadata
 }
 
+fn remap_path(sess: &Session, path: &str) -> String {
+    for (old, new) in sess.opts.debug_prefix_map.clone() {
+        if path.starts_with(&old) {
```

Comment:
Are `old` and `new` guaranteed to end with path separator characters, or will this end up doing arbitrary remapping within the middle of a path?